[Test plan][Smartswitch]Test plan for feature ENI Based Forwarding

congh-nvidia opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
This is the test plan for smartswitch feature ENI Based Forwarding.
Feature HLD: 
https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/eni-based-forwarding.md
Test PR: https://github.com/sonic-net/sonic-mgmt/pull/21251


### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ ] Test case improvement


### Back port request
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?

#### Any platform specific information?
Only for smartswitch platforms.
#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

congh-nvidia committed:
Create Dash-ENI-Based-Forwarding-test-plan.md

mssonicbld commented:
/azp run

congh-nvidia commented:
@vivekrnv  @oleksandrivantsiv @nhe-NV could you please review?

congh-nvidia committed:
Update Dash-ENI-Based-Forwarding-test-plan.md

Fix typo.

mssonicbld commented:
/azp run

roy-sror commented:
@wangxin , @yxieca  - can you pls help to review this change

congh-nvidia commented:
Hi @roy-sror , we will have a test plan review for this on the next SmartSwitch Group Meeting(should be this week).

congh-nvidia committed:
Update Dash-ENI-Based-Forwarding-test-plan.md

mssonicbld commented:
/azp run

r12f commented on docs/testplan/dash/Dash-ENI-Based-Forwarding-test-plan.md:
```diff
@@ -0,0 +1,163 @@
+
+
+## Test
+### Test case # 1 – test_privatelink_basic_transform migrate to ENI based fowarding
```
Hi @congh-nvidia , after a second thought, we don't have to create a fixture/plugin for reusing the test cases. We can simply call the existing test functions for sending and validating the traffic. In this way, the top level test cases don't have to change, and also easy for future changes, say from DPU to ENI.

congh-nvidia replied:
Hi @r12f , according to the design changed in https://github.com/sonic-net/SONiC/pull/2081 and discussion we had with MSFT team, I have update the test plan and narrowed the test scope to only test on the NPU. The DPU dataplane interface is mocked by a NPU front panel interface.
In this case, we don't need consider the other existing dash tests and the functionality of this feature can be fully  covered in this test.
Please review the test plan, the test is also ready, I will create the PR soon.

congh-nvidia committed:
Update Dash-ENI-Based-Forwarding-test-plan.md

Update the test plan due to the design change.

mssonicbld commented:
/azp run

congh-nvidia committed:
Update Dash-ENI-Based-Forwarding-test-plan.md

Update the test plan due to the design change: https://github.com/sonic-net/SONiC/pull/2081

congh-nvidia commented:
Hi @prabhataravind @vvolam could you please help review this?

mssonicbld commented:
This PR has backport request label(s) for branch(es): 202511, but is missing required test information. Please make sure you tick the tested branch(es) in the **Tested branch** section and provide test evidence (e.g., 202511: &lt;test result&gt;) in the **Test result** section as well in your PR description.

<div align="right"><sub>---Powered by SONiC BuildBot</sub></div>
